### PR TITLE
fixed double free happening during EvictSessionFromCache

### PR DESCRIPTION
**Pull request by SKlimaRA**
# Description

Fixes double free.

Fixes zd#17293

# Testing

How did you test?

# Checklist

 - [ ] added tests
 - [ ] updated/added doxygen
 - [ ] updated appropriate READMEs
 - [ ] Updated manual and documentation

### CODE REVIEW

**Comment by dgarske** [comment]
Okay to test. Approved contributor

**Review by SparkiDev — DISMISSED** [review]

File: src/ssl.c [hunk 1]
```diff
@@ -19774,6 +19774,7 @@ void wolfSSL_FreeSession(WOLFSSL_CTX* ctx, WOLFSSL_SESSION* session)
 #ifdef HAVE_SESSION_TICKET
     if (session->ticketLenAlloc > 0) {
         XFREE(session->ticket, session->heap, DYNAMIC_TYPE_SESSION_TICK);
+        session->ticketLenAlloc = 0;
```
Comment (julek-wolfssl) [inline src/ssl.c:19777]: Let's also clear `session->ticket`.

**Review by julek-wolfssl — CHANGES_REQUESTED** [review]

File: src/ssl.c (hunk 1 above)
Comment (julek-wolfssl) [reply src/ssl.c:19777]: Also `session->ticketNonce.data` is not being cleared. Please clear that too.

**Comment by julek-wolfssl** [comment]
The curl test is failing due to a missing dependency. Please rebase your PR on latest master to pick up the fix for this test.

File: src/ssl.c (hunk 1 above)
Comment (SKlimaRA) [reply src/ssl.c:19777]: done, i hope

Comment (julek-wolfssl) [reply src/ssl.c:19777]: Great work!

**Review by julek-wolfssl — APPROVED** [review]

**Review by SparkiDev — APPROVED** [review]

**Pull request merged**